src/v1/parser.rs

```diff
@@ -0,0 +1,814 @@
+use std;
+use std::borrow::Cow;
+use std::convert::From;
+use std::error::Error;
+use std::ffi::{OsStr, OsString};
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::fmt;
+use std::io;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+use std::slice::Chunks;
+use std::str::{FromStr, Utf8Error};
+
+use quick_error::ResultExt;
+
+use ::HashType;
+use super::writer::{MAGIC, VERSION};
+
+
+macro_rules! itry {
+    ($x: expr) => {
+        match $x {
+            Err(e) => return Some(Err(From::from(e))),
+            Ok(v) => v,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ParseRowError(String);
```

Comment:
This should be enum with quick_error. Some options like version mismatch might be useful to handle programmatically.